src/components/expression/expression-loader.js

```diff
@@ -132,7 +133,7 @@ genotet.ExpressionLoader.prototype.loadExpressionMatrix_ = function(fileName,
       });
     this.data.lowerConditionNames = lowerConditionNames;
 
-    this.loadTfaProfile_(fileName, geneNames, conditionNames);
+    this.signal('loadTfaProfile');
```

Comment:
It seems that you can loadTfaProfile_() directly. The signal handler you have in expression.js can be done here.